test/boost/secondary_index_test.cc

```diff
@@ -1560,6 +1560,305 @@ SEASTAR_TEST_CASE(test_token_order) {
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) }
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck1 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck1 = 1 AND ck2 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck2 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) }
+            });
+        };
+
+        // Do tests for each column (excluding partition key columns - cannot mix partition column 
```

Comment:
I would have moved this comment to the *top* or *above* the test. There is no way to guess that this is what this test is actually doing, before going 100 lines into its code.